HW_6/chess/queen.py

```diff
@@ -0,0 +1,54 @@
+# Добавьте в пакет, созданный на семинаре шахматный модуль.
+# Внутри него напишите код, решающий задачу о 8 ферзях.
+# Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они
+# не били друг друга. Вам дана расстановка 8 ферзей на доске, определите,
+# есть ли среди них пара бьющих друг друга. Программа получает на вход восемь
+# пар чисел, каждое число от 1 до 8 - координаты 8 ферзей. Если ферзи не бьют
+# друг друга верните истину, а если бьют - ложь.
+
+# Напишите функцию в шахматный модуль. Используйте генератор случайных чисел
+# для случайной расстановки ферзей в задаче выше. Проверяйте различные случайные
+# варианты и выведите 4 успешных расстановки. *Выведите все успешные варианты
+# расстановок
+
+import random
+
+def check_queen(position: list) -> bool:
+    n = 8
+    x = []
+    y = []
+
+    for i in range(n):
+        x.append(position[i][0])
+        y.append(position[i][1])
+    correct = True
+    for i in range(n):
+        for j in range(i + 1, n):
+            if x[i] == x[j] or y[i] == y[j] or abs(x[i] - x[j]) == abs(y[i] - y[j]):
+                correct = False
+    if correct:
+        return True
+    else:
+        return False
```

Comment:
Зачем нужно разделять двумерный массив на отдельные списки? Можно ведь  сделать так
```suggestion
    for i, (x1, y1) in enumerate(position, 1):
        for x2, y2 in position[i:]:
            if x1 == x2 or y1 == y2 or abs(x1 - x2) == abs(y1 - y2):
                return False

    return True
```